py/lexer.c

```diff
@@ -577,12 +579,20 @@ void mp_lexer_to_next(mp_lexer_t *lex) {
     vstr_reset(&lex->vstr);
 
     // skip white space and comments
-    bool had_physical_newline = skip_whitespace(lex, false);
+    // set the newline tokens at the line and column of the preceding line:
+    // only advance on the pointer until a new line is crossed, save the
+    // line and column, and then readvance it
+    bool had_physical_newline = skip_whitespace(lex, true);
 
     // set token source information
     lex->tok_line = lex->line;
     lex->tok_column = lex->column;
 
+    if (had_physical_newline) {
+        next_char(lex);
```

Comment:
I don't think this line is needed, the char is anyway consumed in the following call to `skip_whitespace()`.